pkg/util/utils.go

```diff
@@ -178,3 +179,17 @@ func CommandExecutesSuccessfully(cmd string, args ...string) bool {
 	}
 	return false
 }
+
+// IsDirectoryWritable checks if the directory is writable or not
+// by trying creating a temporary file
+// Return true if directory is writable else false
+func IsDirectoryWritable(path string) bool {
```

Comment:
Couldn't find a direct logic hence using it. Let me know if have good solution.

Reply:
I think this is good, it finds out the truth if the directory is actually wirtable or not.
I was thinking about using `Stat()` and get the `Mode().Perm()` but this will only work in unix like systems.
> IsReadable and IsWritable wouldn't be reliable enough... the filesystem can be mounted read-only, or corrupt, or there could be extended attribute ACLs on the file, etc.  The only reliable strategy is to just try to open it for read or write and see if it fails.


Found this mail threaed on the golang-dev mailing list abou it: https://groups.google.com/forum/#!msg/golang-dev/qfenPFNYdYY/8tkDvE3KaUQJ

Reply:
If `Brad Fitzpatrick` is saying then I am sure it is one of good approach :)